python/pyspark/serializers.py

```diff
@@ -401,6 +428,23 @@ def __repr__(self):
         return "ArrowStreamPandasUDFSerializer"
 
 
+class PandasCogroupSerializer(ArrowStreamPandasUDFSerializer):
```

Comment:
Maybe rename this to "CogroupUDFSerializer"? It matches with `ArrowStreamPandasUDFSerializer` a little better.